cast/java/test/data/build.gradle.kts

```diff
@@ -59,3 +59,7 @@ sourceSets.test.get().java.srcDir(downloadJLex.map { it.extra["downloadedSourceD
 //
 
 tasks.register("prepareMavenBuild") { dependsOn("eclipseClasspath", "eclipseProject") }
+
+// On JDK 17, deprecation errors in ECJ cannot be disabled when compiling JLex code.  So, we disable
+// the ECJ task.
+tasks.named("compileTestJavaUsingEcj") { enabled = false }
```

Comment:
@liblit this is a JDK 17 specific issue, and I have no idea how to fix it.  If you have a better thought here let me know.  If you like we can conditionally disable this task only on JDKs with version greater than 11, or something like that.

Reply:
I prefer to keep this task enabled on JDKs where we can make it work. There's still some value in the extra checks that ECJ will apply to those builds.

In consideration of your eagerness to move this pull request forward, I will approve it in its current form. However, I would appreciate it if you could follow up with a pull request that restores this task in at least some subset of our builds. Thanks!

Reply:
I have limited disabling the task to JDK 17+: bc4a19c26f863f03a832d217a287d04f778b5664